metaflow/plugins/argo/argo_events_decorator.py

```diff
@@ -0,0 +1,154 @@
+                        "*name* key."
+                    )
+                self.triggers.append(self.attributes["event"])
+            else:
+                raise MetaflowException(
+                    "Incorrect format for *event* attribute in *@trigger* decorator. "
+                    "Supported formats are string and dictionary - \n"
+                    "@trigger(event='foo') or @trigger(event={'name': 'foo', "
+                    "'parameters': {'alpha': 'beta'}})"
+                )
+        elif self.attributes["events"]:
+            # events attribute supports the following formats -
+            #     1. events=[{'name': 'table.prod_db.members',
+            #               'parameters': {'alpha': 'member_weight'}},
+            #                {'name': 'table.prod_db.metadata',
+            #               'parameters': {'beta': 'grade'}}]
+            if isinstance(self.attributes["events"], list):
+                for event in self.attributes["events"]:
+                    if is_stringish(event):
+                        self.triggers.append({"name": str(event)})
+                    elif isinstance(event, dict):
+                        if "name" not in dict(event):
+                            raise MetaflowException(
+                                "One or more events in *events* attribute for "
+                                "*@trigger* are missing the *name* key."
+                            )
+                        self.triggers.append(event)
+                    else:
+                        raise MetaflowException(
+                            "One or more events in *events* attribute in *@trigger* "
+                            "decorator have an incorrect format. Supported format "
+                            "is dictionary - \n"
+                            "@trigger(events=[{'name': 'foo', 'parameters': {'alpha': "
+                            "'beta'}}, {'name': 'bar', 'parameters': "
+                            "{'gamma': 'kappa'}}])"
+                        )
+            else:
+                raise MetaflowException(
+                    "Incorrect format for *events* attribute in *@trigger* decorator. "
+                    "Supported format is list - \n"
+                    "@trigger(events=[{'name': 'foo', 'parameters': {'alpha': "
+                    "'beta'}}, {'name': 'bar', 'parameters': "
+                    "{'gamma': 'kappa'}}])"
+                )
+
+        if not self.triggers:
+            raise MetaflowException("No event(s) specified in *@trigger* decorator.")
+
+        # same event shouldn't occur more than once
+        names = [x["name"] for x in self.triggers]
+        if len(names) != len(set(names)):
+            raise MetaflowException(
+                "Duplicate event names defined in *@trigger* decorator."
+            )
+
+
+class TriggerOnFinishDecorator(FlowDecorator):
+    name = "trigger_on_finish"
+    defaults = {
```

Comment:
This needs to change to be consistent with way of specifying branch/project.

Reply:
Let's chat about this. I created a version with `prod`, `user` etc. but thinking through what to specify as the value would be quite confusing for the user. This notion of specifying project and branch already maps to the values in system tags and current object.